File: 35.py
n=1000000
arr=map(lambda x:0,range(0,n))
arr[0]=1
arr[1]=1
for i in range(2,n):
    if arr[i]==0:
        temp=i*2
        while temp<n:
            arr[temp]=1;
            temp+=i
prime=[]
for i in range(1,n):
    if arr[i]==0:
        prime.append(i)
palin=[]
for pr in prime:
    temp=pr
    count=0
    check=True
    while temp>0 and check==True:
        count=count+1
        if temp%10==2 and temp!=2:
            check=False
        if temp%10==5 and temp!=5:
            check=False
        temp=temp/10
    if check==False:
        continue
    temp=pr
    co=count
    chh=[]
    while co >0 and check==True:
        co=co-1
        ch=temp%10
        temp=temp/10
        temp=temp+ch*(10**(count-1))
        if temp not in prime:
            check=False
        elif temp not in chh :
            chh.append(temp)
    if check==True:
        for k in chh:
            palin.append(k)
print "numbers are %s"%(len(palin))

File: 35.cpp
#include<iostream>
#include<vector>
using namespace std;
int power(int n,int po){
    int sum=1;
    for(int i=0;i<po;i++)
        sum*=n;
    return sum;
}
int main(){
    int len=1000000;
    int a[len];
    for(int i=0;i<len;i++)
        a[i]=0;
    for(int i=2;i<len;i++)
    {
        int temp=i*2;
        if(a[i]==0)
            while(temp<len){
                a[temp]=1;
                temp=temp+i;
            }
    }
    int count=0;
    for(int i=2;i<len;i++){
        int temp=i;
        int po=0;
        bool ch=true;
        while(temp>0){
            if(temp%2==0 || temp%5==0){
                ch=false;
                break;
            }
            temp=temp/10;
            po++;
        }
        if(!ch)
            continue;
        temp=i;
        int it=po;
        int count_temp=0;
        while(it>0 && ch){
            it--;
            temp=temp/10+(temp%10)*power(10,po-1);
            if(a[temp]==0){
                count_temp++;
                a[temp]=1;
            }
            else{
                count_temp=0;
                ch=false;
            }
        }
        count=count+count_temp;
        if(ch==true)
            cout<<"number is "<<i;
    }
    cout<<"count is "<<count+3;
    //2,5,11 are missed by the algo and added to anser deliberately
    return 0;
}
